model-optimizer/extensions/front/kaldi/add_reshape_around_pooling.py

```diff
@@ -49,38 +49,41 @@ def pattern(self):
     @staticmethod
     def replace_pattern(graph: Graph, match: dict):
         node = match['pool']
+        node_name = node.soft_get('name', node.id)
 
         if node.pool_step is None:
             node.stride = int64_array([1, 1, node.window[-1], node.window[-1]])
 
         # create Reshape before convolution
         # shape = [in_shape[0], in_shape[1]/patch_stride, 1, patch_stride]
-        shape = Shape(graph, {}).create_node()
+        shape = Shape(graph, {'name': node_name + '/Shape'}).create_node()
         shape.in_port(0).connect(node.in_port(0).get_source())
 
-        split = create_op_with_const_inputs(graph, VariadicSplit, {1: int64_array(0), 2: int64_array([1, -1])}, {'out_ports_count': 2}, shape)
-        node_pool_stride = Const(graph, {'value': int64_array([node.pool_stride])}).create_node()
-        pow_node = create_op_node_with_second_input(graph, Pow, int64_array([-1]))
+        split = create_op_with_const_inputs(graph, VariadicSplit, {1: int64_array(0), 2: int64_array([1, -1])},
+                                            {'name': shape.name + '/split_batch', 'out_ports_count': 2}, shape)
+
+        pow_node = create_op_node_with_second_input(graph, Pow, int64_array([-1]), {'name': node_name + '/pool_stride/inverse'})
+        node_pool_stride = Const(graph, {'value': int64_array([node.pool_stride]),
+                                         'name': node_name + '/pool_stride/'}).create_node()
         pow_node.in_port(0).connect(node_pool_stride.out_port(0))
 
-        mul = Mul(graph, {}).create_node()
+        mul = Mul(graph, {'name': node_name + '/mul_inverse_stride_h'}).create_node()
         mul.in_port(0).connect(split.out_port(1))
         mul.in_port(1).connect(pow_node.out_port(0))
 
-        const_1 = Const(graph, {'value': int64_array([1])}).create_node()
+        concat = create_op_with_const_inputs(graph, Concat, {2: int64_array([1])},
+                                             {'name': node_name + '/concat_all_dims', 'in_ports_count': 4, 'axis': 0})
 
-        concat = Concat(graph, {'in_ports_count': 4, 'axis': 0}).create_node()
         concat.in_port(0).connect(split.out_port(0))
         concat.in_port(3).connect(mul.out_port(0))
-        concat.in_port(2).connect(const_1.out_port(0))
         concat.in_port(1).connect(node_pool_stride.out_port(0))
 
-        reshape_in = Reshape(graph, {'name': '/Reshape/' + node.name}).create_node()
+        reshape_in = Reshape(graph, {'name': '/Reshape/' + node_name}).create_node()
```

Comment:
Please, change name to `node_name + '/reshape_in'`

Reply:
Done.